packages/salesforcedx-vscode-soql/src/queryResultsView/queryDataViewService.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { JsonMap } from '@salesforce/ts-types';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { getDocumentName } from '../commonUtils';
+import {
+  DATA_VIEW_MEDIA_PATH,
+  QUERY_DATA_VIEW_PANEL_TITLE,
+  QUERY_DATA_VIEW_SCRIPT_FILENAME,
+  QUERY_DATA_VIEW_STYLE_FILENAME,
+  QUERY_DATA_VIEW_TYPE,
+  TABULATOR_SCRIPT_FILENAME,
+  TABULATOR_STYLE_FILENAME
+} from '../constants';
+import { HtmlUtils } from '../editor/htmlUtils';
+import { html } from './queryDataHtml';
+
+export class QueryDataViewService {
+  public currentPanel: vscode.WebviewPanel | undefined = undefined;
+  public readonly viewType = QUERY_DATA_VIEW_TYPE;
+  public static extensionPath: string;
+
+  constructor(
+    private subscriptions: vscode.Disposable[],
+    private queryData: JsonMap[],
+    private document: vscode.TextDocument
+  ) {}
+
+  public static register(context: vscode.ExtensionContext): void {
+    QueryDataViewService.extensionPath = context.extensionPath;
+  }
+
+  private updateWebviewWith(webview: vscode.Webview, queryData: JsonMap[]) {
+    webview.postMessage({
+      type: 'update',
+      data: queryData,
+      documentName: getDocumentName(this.document)
+    });
```

Comment:
This is different than a `CustomEditorEvent` and we only have one post message to the webview, so I left this as a string literal. I figured we can add some constants or enumerations if the messaging grows in the future.